bin/activate-hermit

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+# This file must be used with "source bin/activate-hermit" from bash or zsh.
+# You cannot run it directly
+
+if [ "${BASH_SOURCE-}" = "$0" ]; then
+  echo "You must source this script: \$ source $0" >&2
+  exit 33
+fi
+
+BIN_DIR="$(dirname "${BASH_SOURCE[0]:-${(%):-%x}}")"
```

Comment:
What is this `${(%):-%x}` stuff? It is not valid bash from what I know. Perhaps some zsh has slipped into a bash script?

Reply:
no idea... also autogenerated.

Reply:
I think I know. It is a zsh-ism. This is not run with bash (even though it starts with #!/bin/sh - that is ignored) as it is sourced, it is run by whatever shell you use. BASH_SOURCE will be defined with bash so that funny thing that doesn't work in bash wont be run. But on zsh, BASH_SOURCE will not be set and presumably that funny thing does the same thing for zsh.